### PR TITLE
Add tests and Github actions

### PR DESCRIPTION
**Pull request by Dampolo03**
## Done
- Added more tests for CLI commands.
- Added workflow for Github actions.

## QA
- Check that all tests pass by running `yarn dev doctor`.
- Check that Github actions is properly configured.

## Issue / Card
- [workspaceengineering-bot#22](https://github.com/canonical/workspaceengineering-bot/issues/22)
- [canonical/ght#160](https://github.com/canonical/ght/issues/160)
- [canonical/workplace-engineering-squad#851](https://github.com/canonical/workplace-engineering-squad/issues/851)

## Note
`CREDENTIALS` environment variable should be added to actions secrets.

### CODE REVIEW

File: src/test-greenhouse.ts [hunk 1]
```diff
@@ -54,8 +96,61 @@ async function testDeleteJobPost(sso: SSO, spinner: Ora) {
     }
 }
 
+async function multipleSuccessfulTestReplicateJobPost(sso: SSO, spinner: Ora) {
+    console.log(
+        "Testing multiple job post replicate - this should be successful as regions 'apac'  and 'americas' exist.."
+    );
+    try {
+        await provideAuthentication(sso, (page) =>
+            addPosts(spinner, false, JOB_POST_ID, ["apac", "americas"], page)
```
Comment (anthonydillon) [inline src/test-greenhouse.ts:105]: Could you test all the regions here to ensure they all work going forward?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
One small comment is the code

**Comment by anthonydillon** [comment]
Can you ensure you add an issue for assign tests and interactive tests please.

**Comment by Dampolo03** [comment]
> Can you ensure you add an issue for assign tests and interactive tests please.

@anthonydillon Done.

**Comment by anthonydillon** [comment]
@Dampolo03 I am getting the following error now:
```
Failed to get some information from Greenhouse: Error: Cannot find name of the job with 1753300 ID in the https://canonical.greenhouse.io/plans/1753300 page.
```

**Comment by Dampolo03** [comment]
> @Dampolo03 I am getting the following error now:
> 
> ```
> Failed to get some information from Greenhouse: Error: Cannot find name of the job with 1753300 ID in the https://canonical.greenhouse.io/plans/1753300 page.
> ```

@anthonydillon Okay, I'll try to look into this. Weirdly, it works for me.

**Comment by anthonydillon** [comment]
I just don't have permission to see this role. Let's merge and see if the action works.

**Pull request merged**